taskfiles/utils.yml

```diff
@@ -44,21 +48,24 @@ tasks:
     vars:
       TMP_CHECKSUM_FILE: "{{.CHECKSUM_FILE}}.tmp"
     requires:
-      vars: ["CHECKSUM_FILE", "DATA_DIR"]
+      vars: ["CHECKSUM_FILE", "DATA_PATTERNS"]
     cmds:
       - task: "compute-checksum"
         vars:
-          DATA_DIR: "{{.DATA_DIR}}"
-          EXCLUDE_PATHS:
-            ref: "default (list) .EXCLUDE_PATHS"
+          DATA_PATTERNS:
+            ref: ".DATA_PATTERNS"
+          EXCLUDE_PATTERNS:
+            ref: "default (list) .EXCLUDE_PATTERNS"
           OUTPUT_FILE: "{{.TMP_CHECKSUM_FILE}}"
       - defer: "rm -f '{{.TMP_CHECKSUM_FILE}}'"
-      # Check that the directory exists and the checksum matches; otherwise delete the checksum file
+      # Check that the paths exist and the checksum matches; otherwise delete the checksum file.
       - >-
         (
-        test -d "{{.DATA_DIR}}"
-        && diff -q '{{.TMP_CHECKSUM_FILE}}' '{{.CHECKSUM_FILE}}' 2> /dev/null
-        ) || rm -f '{{.CHECKSUM_FILE}}'
+        {{- range .DATA_PATTERNS}}
+        test -e "{{.}}" &&
```

Comment:
(Ignoring the quotes) I probably should've anticipated this, but this won't work for patterns since if a wildcard path glob returns multiple files, `test -e` will fail because it only expects one argument.